documents/Design.md

```diff
@@ -0,0 +1,25 @@
+# Dataset Viewer Design
+
+Programming Language: Python 3  
+Coding Style: PEP8  
+Design Pattern: Model View Presenter (Passive View)
+## Class Diagram
+![Class Diagram](ClassDiagram.png)  
+DimensionView: Contains the slider, buttons, and stepper for an individual dimension.  
+DimensionPresenter: Manages the behavior of the DimensionView.  
+MainView: Contains the individual DimensionViews and the central plot.  
+MainPresenter: Manages the behavior of the MainView and retrieves data from the MainModel.  
+MainModel: Contains the data array. 
+## Sequence Diagrams
+### Pressing an X Button
+![X Button Press Sequence Diagram](XButtonPress.png)
+### Stepper Change
+![Stepper Change Sequence Diagram](StepperChange.png)
+## Development Principles
+* Testing must take place alongside development
```

Comment:
I will go further and say that ALL development will be done following test driven development.

Reply:
All presenters and tests will be created prior to the development of any concrete views. Putting the Qt widgets together will be the very final step - comes even after testing.